### PR TITLE
Feat/apps navigation

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix MANAGER-5847
| License          | BSD 3-Clause

## Description

* Add `applicationURLs` in `Environment` (require `/configuration#7`)
* add `url-builder` in `@ovh-ux/ufrontend`

Exemple on how to build an url to an external application: 

```js 
import { buildURL } from '@ovh-ux/ufrontend/url-builder';

const url = buildURL('hub', '#/catalog', { q: 'domaine' });
console.log(url);
// https://www.ovh.com/manager/#/catalog?q=domaine
```

Replace #3993

### CODE REVIEW

File: packages/manager/modules/config/src/environment/environment.service.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ export default class EnvironmentService {
     this.userLocale = findAvailableLocale(detectUserLocale(), this.region);
     this.version = null;
     this.user = {};
+    this.baseUrls = {};
```
Comment (frenautvh) [inline packages/manager/modules/config/src/environment/environment.service.js:14]: maybe ```applicationURLs``` would describe the attribute better, what do you think?

**Review by frenautvh — CHANGES_REQUESTED** [review]
i think we need to talk more about it, i'm not sure that we should base routing on state identifiers because it does make sense only in the context of some routers (ui router for example)

take a look at https://reactrouter.com/web/guides/quick-start the routing is resolved using URLs, the concept of state id is not necessary. So in your example you would do '#/goto/app/dashboard' and perform an un-necessary redirection in the react app and it would just be simpler to just do appURL + /app/dashboard

in my opinion the inter-app routing should only rely on crafting URLs, there is no concept of state

File: packages/manager/modules/config/src/environment/environment.service.js (hunk 1 above)
Comment (cbourgois) [reply packages/manager/modules/config/src/environment/environment.service.js:14]: Make sense, it's applications bases urls.

**Comment by cbourgois** [comment]
> i think we need to talk more about it, i'm not sure that we should base routing on state identifiers because it does make sense only in the context of some routers (ui router for example)
> 
> take a look at https://reactrouter.com/web/guides/quick-start the routing is resolved using URLs, the concept of state id is not necessary. So in your example you would do '#/goto/app/dashboard' and perform an un-necessary redirection in the react app and it would just be simpler to just do appURL + /app/dashboard
> 
> in my opinion the inter-app routing should only rely on crafting URLs, there is no concept of state

In fact two way to think about it : 
- state doesn't change and url will change (using goto)
- state will change but urls will remain the same (using baseUrl + urlPart as constant)

So we have to choose one way (both are good but we need to keep one ;) )

If we use urlPart, `buildUrl` will be simplier : 

```js
const buildUrl = (app, url) => {
  return `${Environment.getBaseUrl(app)}${url}`;
};
```

**Comment by cbourgois** [comment]
Ok, `state name` could change and `URL` too, so we will use `URL`.
If URL changes the application that handle should provide a redirection (a customer who has a bookmark on old URL should be redirected to the new one)

Build URL should be simpler : 
```js
const buildUrl = (application, url) => `${Environment.getApplicationURL(application)}${url}`
```

File: packages/manager/modules/ufrontend/url-builder.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Environment } from '@ovh-ux/manager-config';
+
+const buildQueryString = (data) =>
+  Object.keys(data)
+    .map((key) => `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`)
+    .join('&');
+
+export const buildUrl = (application, url, query) => {
```
Comment (frenautvh) [inline packages/manager/modules/ufrontend/url-builder.js:8]: it's really nitpicking but buildURL would be more consistent (encodeURIComponent for example)

**Review by frenautvh — DISMISSED** [review]

File: packages/manager/modules/ufrontend/README.md [hunk 3]
```diff
@@ -114,3 +114,43 @@ emit(
...
 ```
+
+### URL Builder
+
+URL builder is here to help us to build links between applications.
+When you want to build an URL for an outside application, you will use `url-builder` to build the URL.
+
+#### Examples
+
+Build the url that redirect to `#/catalog` state from the `hub` application with query parameter (`expand`)
```
Comment (antleblanc) [inline packages/manager/modules/ufrontend/README.md:125]:
```suggestion
Build the URL that redirect to `#/catalog` state from the `hub` application with query parameter (`expand`)
```

**Review by antleblanc — APPROVED** [review]

**Review by marie-j — APPROVED** [review]

**Review by Jisay — DISMISSED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**